internal/server/submit_download.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"errors"
+	"github.com/opg-sirius-finance-admin/internal/api"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"github.com/opg-sirius-finance-admin/internal/util/util"
+	"net/http"
+)
+
+type SubmitDownloadHandler struct {
```

Comment:
Strange name. The user is requesting a report to be downloaded as a file, not submitting a download.

Reply:
changed